slippy/tile.go

```diff
@@ -2,77 +2,123 @@ package slippy
 
 import (
 	"errors"
-	"fmt"
+	"math"
+	"strconv"
 
 	"github.com/go-spatial/geom"
 )
 
+var (
+	ErrNilBounds = errors.New("slippy: Bounds cannot be nil")
+)
+
 // MaxZoom is the lowest zoom (furthest in)
 const MaxZoom = 22
 
-// NewTile returns a Tile of Z,X,Y passed in
-func NewTile(z, x, y uint) *Tile {
-	return &Tile{
+// Zoom represents a zoom level; this usually goes from 0 to 22
+type Zoom uint
+
+func (z Zoom) N() float64 { return math.Exp2(float64(z)) }
+func (z Zoom) TileSize() Tile {
+	n := int(z.N())
+	return Tile{
 		Z: z,
-		X: x,
-		Y: y,
+		X: n,
+		Y: n,
 	}
 }
 
 // Tile describes a slippy tile.
 type Tile struct {
 	// zoom
-	Z uint
+	Z Zoom
 	// column
-	X uint
+	X int
 	// row
-	Y uint
+	Y int
 }
 
-// NewTileMinMaxer returns the smallest tile which fits the
-// geom.MinMaxer. Note: it assumes the values of ext are
-// EPSG:4326 (lng/lat)
-func NewTileMinMaxer(g Grid, ext geom.MinMaxer) (*Tile, bool) {
-	tile, ok := g.FromNative(MaxZoom, geom.Point{
-		ext.MinX(),
-		ext.MinY(),
-	})
-
-	if !ok {
-		return nil, false
+func (tile Tile) ZXY() (Zoom, int, int) { return tile.Z, tile.X, tile.Y }
+
+// Equal tests for equality
+func (tile Tile) Equal(other Tile) bool {
+	return tile.Z == other.Z && tile.X == other.X && tile.Y == other.Y
+}
+func (tile Tile) Less(other Tile) bool {
```

Comment:
Consider adding a code comment here. This function considers Z, then X then Y when deciding if another tile is Less